card_game.py

```diff
@@ -0,0 +1,214 @@
+# Import required modules
+import random
+from abc import ABC, abstractmethod
+
+# Define Card classes
+
+
+class Card(ABC):
+    def __init__(self, name, ap, info):
+        self.name = name
+        self.ap = ap
+        self.info = info
+
+    @abstractmethod
+    def play(self, player, target):
+        pass
+
+
+class AttackCard(Card):
+    def __init__(self, name, ap, damage, info):
+        super().__init__(name, ap, info)
+        self.damage = damage
+
+    def play(self, player, target):
+        target.take_damage(self.damage)
+
+
+class DefenseCard(Card):
+    def __init__(self, name, ap, defense_point, info):
+        super().__init__(name, ap, info)
+        self.defense_point = defense_point
+
+    def play(self, player, target):
+        player.add_defense(self.defense_point)
+
+
+class DrawCard(Card):
+    def __init__(self, name, ap, draw_count, info):
+        super().__init__(name, ap, info)
+        self.draw_count = draw_count
+
+    def play(self, player, target=None):
+        player.draw_additional_cards(self.draw_count)
+
+# Define Player and Enemy classes
+
+
+class Character:
+    def __init__(self, name, max_hp, ap=3, deck=None):
+        self.name = name
+        self.max_hp = max_hp
+        self.current_hp = max_hp
+        self.ap = ap
+        self.deck = deck if deck else []
+        self.max_hand = 5
+        self.hand_card = []
+        self.used_card = []
+        self.defense_point = 0
+
+    def take_damage(self, damage):
+        if self.defense_point > 0:
+            damage_after_defense = max(0, damage - self.defense_point)
+            self.defense_point = max(0, self.defense_point - damage)
+            self.current_hp -= damage_after_defense
+        else:
+            self.current_hp -= damage
```

Comment:
The if-else can be removed, since the code for `if self.defense_point > 0` works for `else` (assuming `self.defense_point >= 0` is always true.)

Reply:
ok

Reply:
Done